### PR TITLE
Add line highlight for breakpoints

### PR DESCRIPTION
**Pull request by KsavinN**
chrome-capture (5)

resolved:
#85

In future:
#86  

I add simple Highline. Signal instance is created in the model of the debugger. Emitting is placed in service and signal is connected in cell handler. Is that proper way or should we consider about another architecture?

### CODE REVIEW

File: src/debugger.ts [hunk 1]
```diff
@@ -172,6 +176,7 @@ export namespace Debugger {
     private _session: IDebugger.ISession | null;
     private _sessionChanged = new Signal<this, void>(this);
     private _service = new DebugService(null, this);
+    private _selectCurrentLine = new Signal<this, number>(this);
```
Comment (jtpio) [inline src/debugger.ts]: How about this name to follow the other conventions?

```suggestion
    private _currentLineChanged = new Signal<this, number>(this);
```

File: style/breakpoints.css [hunk 2]
```diff
@@ -13,3 +13,7 @@
   top: -1px;
   color: #f22;
 }
+
+.highlight {
```
Comment (jtpio) [inline style/breakpoints.css]: We could think of a more descriptive class name.

File: style/breakpoints.css [hunk 3]
```diff
@@ -13,3 +13,7 @@
...
+  background-color: orange;
```
Comment (jtpio) [inline style/breakpoints.css]: Here we could make use of a JupyterLab CSS variable (to also handle the dark theme)

**Comment by jtpio** [comment]
> Emitting is placed in service and signal is connected in cell handler. Is that proper way or should we consider about another architecture?

I think it makes sense to have it there for now and let the service update the model.

Later the service will be able to:

1. listen to `stopped` events
2. retrieve `scopes`, `stackTraces` and `variables` on `stopped`
3. update the model data (`scopes`, `variables`, `currentLine` and so on)

The views would listen to changes on their model and reflect the updates.

**Comment by jtpio** [comment]
Note that this change doesn't remove the line highlighting when the debugger stops.

We'll need some kind of cleanup handler for that.

File: src/handlers/cell.ts [hunk 4]
```diff
@@ -36,6 +43,24 @@ export class CellManager implements IDisposable {
   private _activeCell: CodeCell;
   isDisposed: boolean;
 
+  showCurrentLine(lineNumber: number) {
```
Comment (jtpio) [inline src/handlers/cell.ts]: Could this method actually be private?

**Comment by KsavinN** [comment]
I added a suggestion of @jtpio.
If everything looks ok I guess we can merge that.
Rest development will be go on after PR #93.
@afshin

**Comment by jtpio** [comment]
Going to rebase this one to fix the conflicts.

**Comment by jtpio** [comment]
After getting the latest changes from master (tree view):

line-highlight

**Comment by KsavinN** [comment]
> After getting the latest changes from master (tree view):
> 
> One more step to real debugger :)

**Pull request merged**